frontend/src/components/pages/TranslationPage.tsx

```diff
@@ -73,13 +91,19 @@ const TranslationPage = () => {
     }
   };
 
+  const updateVersionHistory = (content: string, lineIndex: number) => {
```

Comment:
This is very open for discussion, but this abstraction feel strange to me since this "level" of change is kept within the function body in the redo and undo methods and also, the name implies this method would work for the redo/undo scenarios.

Basically, I don't think giving its own method is super helpful. Thoughts?

Reply:
Understandable and I totally see it now, I'll move this functionality into `onUserInput`.